### PR TITLE
add batch requests support for the vmss.

### PR DESCRIPTION
**Pull request by andyliuliming**
1. fix the subnet fields not updated correctly.
2. add batch requests support for the vmss.

### CODE REVIEW

**Comment by andyliuliming** [comment]
@bingosummer  @gossion  ping :)

File: src/bosh_azure_cpi/lib/cloud/azure/cloud.rb [hunk 1]
```diff
@@ -743,9 +743,10 @@ def _init_azure
       cloud_error("Please make sure the CPI has proper network access to Azure. #{e.inspect}") # TODO: Will it throw the error when initializing the client and manager
     end
 
-    def _init_cpi_lock_dir
-      CPILogger.instance.logger.info('_init_cpi_lock_dir: Initializing the CPI lock directory')
+    def _init_cpi_dirs
+      CPILogger.instance.logger.info('_init_cpi_dirs: Initializing the CPI lock directory')
```
Comment (bingosummer) [inline src/bosh_azure_cpi/lib/cloud/azure/cloud.rb]: `Initializing the CPI lock directory` => `Initializing the CPI directory`

File: src/bosh_azure_cpi/lib/cloud/azure/vms/vmss_manager.rb [hunk 2]
```diff
@@ -29,61 +29,60 @@ def create(bosh_vm_meta, vm_props, network_configurator, env)
       vmss_name = _get_vmss_name(vm_props, env)
       _ensure_resource_group_exists(resource_group_name, vm_props.location)
 
-      existing_vmss = @azure_client.get_vmss_by_name(resource_group_name, vmss_name)
-      vmss_params = {}
+      vmss_params = {
+        resource_group_name: resource_group_name
+      }
```
Comment (bingosummer) [inline src/bosh_azure_cpi/lib/cloud/azure/vms/vmss_manager.rb]: It seems that `vmss_params` are re-defined in line 37.

**Comment by bingosummer** [comment]
@andyliuliming , could you please add more details about `batch requests support for the vmss`? E.g., what's the problem it fixed? Why it's supported?
In future, we or others can view the PR message to understand why we made this change. :)

File: src/bosh_azure_cpi/lib/cloud/azure/vms/vmss_manager.rb (hunk 2 above)
Comment (andyliuliming) [reply src/bosh_azure_cpi/lib/cloud/azure/vms/vmss_manager.rb]: resolved.

File: src/bosh_azure_cpi/lib/cloud/azure/cloud.rb (hunk 1 above)
Comment (andyliuliming) [reply src/bosh_azure_cpi/lib/cloud/azure/cloud.rb]: changed.

**Comment by andyliuliming** [comment]
> @andyliuliming , could you please add more details about `batch requests support for the vmss`? E.g., what's the problem it fixed? Why it's supported?
> In future, we or others can view the PR message to understand why we made this change. :)

updated the commit message. thanks.

File: src/bosh_azure_cpi/lib/cloud/azure/batching/batching.rb [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    # grouping_key is somethink like the vmss_name.
+    def _try_to_execute(request, executor)
+      grouping_key = request[:grouping_key]
+      task_in_file_path = _get_task_in_file_path(grouping_key)
+      # handling the items int the fifo identified by grouping_key by executor.
+      task_executor_lock_name = _get_task_executor_lock_name(grouping_key)
+      sleep(3) # sleep 3 seconds to let the task file grow as big as possible.
+      flock(task_executor_lock_name, File::LOCK_EX) do
+        # read all the items in the task file
+        flock(_get_task_in_file_lock_name(grouping_key), File::LOCK_EX) do
```
[the remark this answers is not in the document]
Comment (andyliuliming) [reply src/bosh_azure_cpi/lib/cloud/azure/batching/batching.rb]: it locks different things.
the first is to ensure that only one executor are executing.
the second one is for locking the task_in_file.

**Pull request merged**